docs/tutorial/category_theory/category_theory_intro.md

```diff
@@ -0,0 +1,231 @@
+the objects. The objects and morphisms have some extra structure and satisfy some axioms -- see the
+[definition on Wikipedia](https://en.wikipedia.org/wiki/Category_(mathematics)#Definition) for
+details.
+
+One important thing to note is that a morphism in an abstract category may not be an actual map
+between two types. In particular, there is new notation `⟶` , typed as `\h` in VS Code, for a
+morphism, and in some fonts this arrow can be virtually indistinguishable from the standard function
+arrow `→` . (**TODO** -- perhaps say something about how to change font, or point to a resource? kmb
+would love to know this himself because he cannot tell the difference between the arrows on his
+laptop; one seems to be one pixel bigger than the other). In particular, usual functional
+programming techniques for function evaluations (e.g. writing `f x` if `x : X` and `f : X → Y`) are
+not in general possible when using morphisms in category theory (although sometimes this is possible
+(**TODO** is it?)).
+
+Another point of confusion can be universe issues. Following Lean's conventions for universe
+polymorphism, the objects of a category might live in one universe `u` and the morphisms in another
+universe `v`. Note that in many categories showing up in "set-theoretic mathematics", the morphisms
+between two objects often form a set, but the objects themselves may or may not form a set. In Lean
+this corresponds to the two possibilities `u=v` and `u=v+1`, known as `small_category` and
+`large_category` respectively. In the below we stick to the general polymorphic situation with `u`
+and `v` general universes.
+
+## Getting started with categories
+
+The structure of a category on a type `C` in Lean is done using typeclasses; terms of `C` then
+correspond to objects in the category. The convention in the category theory library is that `C :
+Type u`, and if `X : C` and `Y : C` then morphisms `X ⟶ Y : Type v` (note the non-standard arrow).
+We set this up as follows:
+
+```lean
+import category_theory.category
+
+universes v u -- the order matters
+
+open category_theory
+
+variables (C : Type u) [𝒞 : category.{v} C]
+
+variables (W X Y Z : C)
+
+variables (f : W ⟶ X) (g : X ⟶ Y) (h : Y ⟶ Z)
+
+include 𝒞
+```
+
+This says "let `C` be a category, let `W`, `X`, `Y`, `Z` be objects of `C`, and let `f : W ⟶ X`, `g
+: X ⟶ Y` and `h : Y ⟶ Z` be morphisms in `C` (with the obvious source and targets)". Note two
+unusual things: firstly the typeclass `category C` is explicitly named (in contrast to group theory,
+where one would just write `[group G]` rather than `[h : group G]`), and secondly we have to
+explicitly tell Lean the universe where the morphisms live, because Lean cannot guess from knowing
+`C` alone. The order which universes are introduced also matters. This might initially look a little
+inelegant, but the theory works fine if these conventions are kept in mind. The reason that the
```

Comment:
```suggestion
`C` alone. The order which universes are introduced also matters (the universe level of the objects can nearly always be inferred, so we put that last). This might initially look a little
inelegant, but the theory works fine if these conventions are kept in mind. The reason that the
```